src/mod/server.mod/sasl.c

```diff
@@ -0,0 +1,716 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
+/*
+ * sasl.c -- part of server.mod
+ *
+ * Written by Michael Ortmann
+ *
+ * Copyright (C) 2019 - 2024 Eggheads Development Team
+ */
+
+#undef answer /* before resolv.h because it could collide with src/mod/module.h
+               * (dietlibc) */
+#include <resolv.h> /* base64 encode b64_ntop() and base64 decode b64_pton() */
+
+/* RFC 5802 - printable ASCII characters excluding ','
+ * printable = %x21-2B / %x2D-7E
+ */
+#define CHARSET_SCRAM "\x21\x22\x23\x24\x25\x26\x27\x28\x29\x2a\x2b\x2d\x2e\x2f\x30\x31\x32\x33\x34\x35\x36\x37\x38\x39\x3a\x3b\x3c\x3d\x3e\x3f\x40\x41\x42\x43\x44\x45\x46\x47\x48\x49\x4a\x4b\x4c\x4d\x4e\x4f\x50\x51\x52\x53\x54\x55\x56\x57\x58\x59\x5a\x5b\x5c\x5d\x5e\x5f\x60\x61\x62\x63\x64\x65\x66\x67\x68\x69\x6a\x6b\x6c\x6d\x6e\x6f\x70\x71\x72\x73\x74\x75\x76\x77\x78\x79\x7a\x7b\x7c\x7d\x7e"
```

Comment:
Can we move this into server.h?

Reply:
We could, but i think sasl.c is a better place for this, for its only related to sasl. similar to the defines and enums that follow. But we could move it into function sasl_scram_step_0().

Reply:
how shall i proceed?